cpp/src/reshape/table_to_array.cu

```diff
@@ -0,0 +1,146 @@
+
+  std::vector<void const*> h_srcs(num_columns);
+  std::vector<void*> h_dsts(num_columns);
+
+  for (int i = 0; i < num_columns; ++i) {
+    auto const& col = input.column(i);
+    CUDF_EXPECTS(col.type() == input.column(0).type(), "All columns must have the same dtype");
+    CUDF_EXPECTS(col.null_count() == 0, "All columns must be non-nullable or contain no nulls");
+
+    h_srcs[i] = static_cast<void const*>(col.data<T>());
+    h_dsts[i] = static_cast<void*>(base_ptr + i * item_size * num_rows);
+  }
+
+  CUDF_CUDA_TRY(cudaMemcpyAsync(d_srcs.data(),
+                                h_srcs.data(),
+                                sizeof(void*) * num_columns,
+                                cudaMemcpyHostToDevice,
+                                stream.value()));
+  CUDF_CUDA_TRY(cudaMemcpyAsync(d_dsts.data(),
+                                h_dsts.data(),
+                                sizeof(void*) * num_columns,
+                                cudaMemcpyHostToDevice,
+                                stream.value()));
+
+  thrust::constant_iterator<size_t> sizes(static_cast<size_t>(item_size * num_rows));
+
+  void* d_temp_storage      = nullptr;
+  size_t temp_storage_bytes = 0;
+  cub::DeviceMemcpy::Batched(d_temp_storage,
+                             temp_storage_bytes,
+                             d_srcs.begin(),
+                             d_dsts.begin(),
+                             sizes,
+                             num_columns,
+                             stream.value());
+
+  rmm::device_buffer temp_storage(temp_storage_bytes, stream);
+
+  cub::DeviceMemcpy::Batched(temp_storage.data(),
+                             temp_storage_bytes,
+                             d_srcs.begin(),
+                             d_dsts.begin(),
+                             sizes,
+                             num_columns,
+                             stream.value());
+}
+
+struct TableToArrayDispatcher {
+  table_view const& input;
+  void* output;
+  rmm::cuda_stream_view stream;
+
+  template <typename T, CUDF_ENABLE_IF(is_fixed_width<T>() || is_fixed_point<T>())>
+  void operator()() const
+  {
+    if constexpr (is_fixed_point<T>()) {
+      using StorageType = cudf::device_storage_type_t<T>;
+      _table_to_array<StorageType>(input, output, stream);
+    } else {
+      _table_to_array<T>(input, output, stream);
+    }
+  }
+
+  template <typename T, CUDF_ENABLE_IF(!is_fixed_width<T>() && !is_fixed_point<T>())>
+  void operator()() const
+  {
+    CUDF_FAIL("Unsupported dtype");
+  }
+};
+
+}  // namespace
+
+void table_to_array(table_view const& input,
+                    void* output,
+                    data_type output_dtype,
+                    rmm::cuda_stream_view stream)
+{
+  CUDF_EXPECTS(output != nullptr, "Output pointer cannot be null.");
+  CUDF_EXPECTS(input.num_columns() > 0, "Input must have at least one column.");
+
+  cudf::type_dispatcher(output_dtype, TableToArrayDispatcher{input, output, stream});
```

Comment:
```suggestion
  cudf::type_dispatcher<device_storage_type>(output_dtype, TableToArrayDispatcher{input, output, stream});
```
Use this so you do not need the if `is_fixed_point<T>` clause in the `TableToArrayDispatcher` operator.

Reply:
Thanks